src/entities/group/group.service.ts

```diff
@@ -0,0 +1,70 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import { GroupModel, UserGroupModel } from '../../db/models';
+import { GroupDto, GroupInterface } from './group.interface';
+import { sequelizeConnection } from '../../db';
+
+class GroupService {
+  async getById(groupId: string) {
+    return await GroupModel.findByPk(groupId);
+  }
+
+  async getAll(limit: number) {
+    return await GroupModel.findAll({
+      limit,
+    });
+  }
+
+  async create(groupData: GroupDto): Promise<GroupModel> {
+    const { name, permissions } = groupData;
+
+    const newGroup = new GroupModel();
+    newGroup.id = uuidv4();
+    newGroup.name = name;
+    newGroup.permissions = permissions;
+
+    await newGroup.save();
+
+    return newGroup;
+  }
+
+  async update(groupDto: Partial<GroupInterface>, groupId: string) {
+    const { name, permissions } = groupDto;
+    const groupToUpdate = await GroupModel.findOne({
+      where: {
+        id: groupId,
+      },
+    });
+
+    if (groupToUpdate) {
+      if (name) groupToUpdate.name = name;
+      if (permissions) groupToUpdate.permissions = permissions;
```

Comment:
Why not `GroupModel.update`?  In that case you don't need to specify updating for each field and validation middleware will help you to get rid of unnecessary fields

Reply:
Agree, this is a better option